angular/src/providers/angular-delegate.ts

```diff
@@ -96,14 +98,29 @@ export const attachView = (
   params: any,
   cssClasses: string[] | undefined
 ): any => {
-  const factory = resolver.resolveComponentFactory(component);
+  let componentRef: ComponentRef<any>;
   const childInjector = Injector.create({
     providers: getProviders(params),
     parent: injector,
   });
-  const componentRef = location
-    ? location.createComponent(factory, location.length, childInjector)
-    : factory.create(childInjector);
+
+  if (resolverOrInjector && isComponentFactoryResolver(resolverOrInjector)) {
+    // Angular 13 and lower
+    const factory = resolverOrInjector.resolveComponentFactory(component);
+    componentRef = location
+      ? location.createComponent(factory, location.length, childInjector)
+      : factory.create(childInjector);
```

Comment:
Will we still need this when we drop Angular 13 support? If not, could add the tech debt ticket comment here too.

Reply:
TBD. The standalone components are in dev preview, so Angular is still reserving the right to completely remove them if they discover a reason why they shouldn't exist. As far as I can tell, all of this is backwards support for the previous model of using the component factory resolver for creating the component to resolve for a route. It is still present in Angular's implementation of `router-outlet`, so we would likely want to maintain it until they remove it, which would be Angular 15 at the earliest.